libs/ngxtension/svg-sprite/src/svg-sprite.ts

```diff
@@ -0,0 +1,471 @@
+import { from, map, shareReplay, switchMap, type Observable } from 'rxjs';
+
+/**
+ * Represents a svg sprite.
+ *
+ * @see {@link NgxSvgSpriteFragment.sprite}
+ * @see {@link NgxSvgSprites.register}
+ * @see {@link https://developer.mozilla.org/en-US/docs/Web/SVG/Element/use `use` Element}
+ */
+export interface NgxSvgSprite {
+	/**
+	 * A unique name identifying this sprite.
+	 */
+	name: string;
+
+	/**
+	 * E.g. `path/to/my/sprite.svg`
+	 */
+	baseUrl: string;
+
+	/**
+	 * @param baseUrl reference to this {@link baseUrl}
+	 * @param fragment
+	 * @returns a url pointing to a specified svg in this sprite by using a `fragment` e.g `path/to/my/sprite.svg#${fragment}`
+	 *
+	 * @see {@link https://svgwg.org/svg2-draft/linking.html#URLReference}
+	 */
+	url: (baseUrl: string, fragment: string) => string;
+
+	/**
+	 * Whether to copy the `viewBox` attribute from the `symbol` in the svg sprite.
+	 */
+	autoViewBox?: boolean;
+
+	/**
+	 *
+	 * @param fragment
+	 * @returns a list of classes that are applied to the svg element.
+	 */
+	classes?: (fragment: string) => string[] | string;
+}
+
+export type CreateSvgSpriteOptions = Omit<NgxSvgSprite, 'url'> &
+	Partial<Pick<NgxSvgSprite, 'url'>>;
+
+/**
+ * Creates a {@link NgxSvgSprite}.
+ *
+ * @param options
+ * @returns
+ */
+export const createSvgSprite = (options: CreateSvgSpriteOptions) => {
+	if (options.url == null)
+		options.url = (baseUrl, fragment) => `${baseUrl}#${fragment}`;
+
+	return options as NgxSvgSprite;
+};
+
+/**
+ * This service registers {@link NgxSvgSprite svg sprites}, which can be rendered via {@link NgxSvgSpriteFragment}.
+ */
+@Injectable({ providedIn: 'root' })
+export class NgxSvgSprites {
+	/**
+	 * @ignore
+	 */
+	private readonly ngZone = inject(NgZone);
+
+	/**
+	 * <{@link NgxSvgSprite.name}, {@link NgxSvgSprite}>
+	 */
+	private readonly sprites: Record<
+		string,
+		NgxSvgSprite & { svg$: Observable<SVGGraphicsElement> }
+	> = {};
+
+	/**
+	 * Registers a sprite.
+	 *
+	 * @param sprite
+	 *
+	 * @see {@link NgxSvgSpriteFragment.sprite}
+	 */
+	public readonly register = (sprite: NgxSvgSprite) => {
+		this.sprites[sprite.name] = {
+			...sprite,
+			svg$: from(
+				this.ngZone.runOutsideAngular(() => fetch(sprite.baseUrl)),
+			).pipe(
+				switchMap((response) => response.text()),
+				map((text) => {
+					const svg = new DOMParser()
+						.parseFromString(text, 'text/xml')
+						.querySelector('svg');
+
+					if (svg == null)
+						throw new Error(
+							`[Svg Sprite] the url '${sprite.baseUrl}' does not seem to be a svg.`,
+						);
+
+					return svg;
+				}),
+				shareReplay(1),
+			),
+		};
+	};
+
+	/**
+	 *
+	 * @param name
+	 * @returns a registered sprite by its name or undefined if not registered.
+	 */
+	public readonly get = (name: string) => this.sprites[name];
+}
+
+/**
+ *
+ * @param sprites
+ * @returns an environment provider which registers icon sprites.
+ */
+export const provideSvgSprites = (...sprites: NgxSvgSprite[]) =>
```

Comment:
nit/suggestion: I think `provideSvgSprites([createSvgSprite()])` is a little verbose. Can't we accept literal objects from the consumers and run `createSvgSprite` internally? In other words, keep `createSvgSprite` an internal API (so one less thing the consumers should care about)